File: php/src/Model/WorkflowRunHistoryEntry.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One completed step in a run history.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkflowRunHistoryEntry
{
    public function __construct(
        public readonly string $stepId,
        public readonly string $taskId,
        public readonly string $workerId,
        public readonly int $completedAt,
        /** @var array<string, mixed>|null */
        public readonly ?array $result,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            stepId: (string) ($data['stepId'] ?? ''),
            taskId: (string) ($data['taskId'] ?? ''),
            workerId: (string) ($data['workerId'] ?? ''),
            completedAt: (int) ($data['completedAt'] ?? 0),
            result: $data['result'] ?? null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkflowRunHistoryEntry.java
package io.fivexer.sdk.model;

import java.util.Map;

/**
 * One completed step in a run's history.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkflowRunHistoryEntry {
    private String stepId;
    private String taskId;
    private String workerId;
    private long completedAt;
    private Map<String, Object> result;

    public String getStepId() { return stepId; }
    public String getTaskId() { return taskId; }
    public String getWorkerId() { return workerId; }
    public long getCompletedAt() { return completedAt; }
    public Map<String, Object> getResult() { return result; }
}
